python/pyarrow/types.py

```diff
@@ -54,351 +56,165 @@ def is_null(t):
     return t.id == lib.Type_NA
 
 
+@doc(is_null, datatype="boolean")
 def is_boolean(t):
-    """
-    Return True if value is an instance of a boolean type.
-
-    Parameters
-    ----------
-    t : DataType
-    """
     return t.id == lib.Type_BOOL
 
 
+@doc(is_null, datatype="integer")
```

Comment:
```suggestion
@doc(is_null, datatype="any integer")
```

I would keep the "any" to clearly indicate this is for _any_ integer (i.e. all bitwidths and signed/unsigned)

Reply:
Good call! Will update